lib/utils/resolveObject.js

```diff
@@ -100,9 +97,24 @@ function compile_value(value, stack) {
         // Recursive so we can compute multi-layer variables like a = b, b = c, so a = c
         if ( to_ret.indexOf('{') > -1 ) {
           var reference = to_ret.slice(1, -1);
-          if(stack.indexOf(reference)!==-1) {
-            stack.push(reference);
-            throw new Error('Variable reference for \'' + stack[0] + '\' resolves to circular definition cycle of \'' + reference + '\':  ' + stack.join(', ') );
+
+          // Compare to found circular references
+          var foundCircArr = Object.keys(foundCirc);
+          if (foundCircArr.indexOf(reference)!==-1) {
```

Comment:
We don't need to create an array of object keys, we could just do `foundCircArr.hasOwnProperty(reference)`

Reply:
Good call. Was using it for console output but obviously thats all gone now.